src/main/java/co/helmethair/scalatest/ScalatestEngine.java

```diff
@@ -47,13 +60,54 @@ public void execute(ExecutionRequest executionRequest) {
     }
 
     @SuppressWarnings("unchecked")
-    private List<Class<? extends Suite>> discoverClassSelectors(EngineDiscoveryRequest dicoveryRequest) {
-        return dicoveryRequest.getSelectorsByType(ClassSelector.class).stream().map(ClassSelector::getJavaClass)
-                .filter(c -> !(c.getEnclosingMethod() != null //only local or anonymous classes have an enclosing method
-                        || c.isSynthetic()
-                        || Modifier.isAbstract(c.getModifiers())
-                        || c.getAnnotation(DoNotDiscover.class) != null)
-                        && Suite.class.isAssignableFrom(c)
-                ).map(c -> ((Class<? extends Suite>) c)).collect(Collectors.toList());
+    private List<Class<? extends Suite>> discoverUniqueIdSelectors(EngineDiscoveryRequest discoveryRequest) {
+        return discoveryRequest.getSelectorsByType(UniqueIdSelector.class).stream()
+            .map(this::getSuite)
+            .filter(Optional::isPresent)
+            .map(o -> (Class<? extends Suite>) o.get()).collect(Collectors.toList());
+    }
+
+    private Optional<?> getSuite(UniqueIdSelector u) {
+        UniqueId uniqueId = u.getUniqueId();
+        if (uniqueId.hasPrefix(UniqueId.forEngine(ID))) {
+            UniqueId.Segment segment = uniqueId.getSegments().get(1);
+            if (SUITE_TYPE.equals(segment.getType())) {
+                String suiteName = segment.getValue();
+                return getSuiteClass(suiteName);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @SuppressWarnings("unchecked")
+    private Optional<Class<? extends Suite>> getSuiteClass(String suiteName) {
+        try {
+            // TODO: not great as it's already done in the discover method and error case is handled there...
+            //  this would require refactoring a bit the discover method
+            Class<?> aClass = Class.forName(suiteName);
+            if (isScalaSuite(aClass)) {
+                return Optional.of((Class<? extends Suite>) aClass);
+            }
+        } catch (ClassNotFoundException e) {
+            //TODO: log ?
```

Comment:
Just logging an error or silently catching it could cover errors and make the build pass. It should be surfaced.

Reply:
Yes, I adressed it by moving the logic to the `Discover.discover` that now takes a list of class names. It should also adress all comments related to the generics since now in `ScalatestEngine` only class names are considered.